PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -784,6 +1289,64 @@ HTREEITEM ProjectPanel::addFolder(HTREEITEM hTreeItem, const TCHAR *folderName)
 	return addedItem;
 }
 
+HTREEITEM ProjectPanel::addDirectory(HTREEITEM hTreeItem, const TCHAR *folderName, bool root, const TCHAR *monitorPath, const std::vector<generic_string>* filters)
+{
+	NodeType nodeType(nodeType_dir);
+	int iconindex = INDEX_CLOSED_DIR;
+
+	if (root)
+	{
+		nodeType = nodeType_baseDir;
+		iconindex = ::PathFileExists(monitorPath) ? INDEX_CLOSED_BASEDIR : INDEX_OFFLINE_BASEDIR;
+	}
+
+	const std::vector<generic_string> dummy;
+	const std::vector<generic_string>* finalFilters = filters ? filters : getFilters(hTreeItem);
+	if (finalFilters == nullptr)
+		finalFilters = &dummy;
+	
+
+	HTREEITEM addedItem = _treeView.addItem(folderName, hTreeItem, iconindex, new ProjectPanelData(_directoryWatcher, folderName, monitorPath, nodeType, *finalFilters));
+
+	_treeView.addItem( TEXT(""), addedItem, INDEX_LEAF_DIRFILE, new ProjectPanelData(_directoryWatcher,TEXT(""), TEXT(""), nodeType_dummy ));
+
+	if (getNodeType(hTreeItem) != nodeType_baseDir && getNodeType(hTreeItem) != nodeType_dir)
+	{
+		TreeView_Expand(_treeView.getHSelf(), hTreeItem, TVE_EXPAND);
+	}
+
+	if (getNodeType(hTreeItem) == nodeType_baseDir)
+		_treeView.setItemImage(hTreeItem, INDEX_OPEN_BASEDIR, INDEX_OPEN_BASEDIR);
+
+	return addedItem;
+}
+
+HTREEITEM ProjectPanel::addDirectory(HTREEITEM hTreeItem)
+{
+	if (_selDirOfFilesFromDirDlg == TEXT("") && _workSpaceFilePath != TEXT(""))
+	{
+		TCHAR dir[MAX_PATH];
+		lstrcpy(dir, _workSpaceFilePath.c_str());

```

Comment:
***never, ever, EVER use `lstrcpy`.***

["lstrcpy is a similarly named Microsoft design defect that appears to behave like strlcpy but is actually a security bug."](https://randomascii.wordpress.com/2013/04/03/stop-using-strncpy-already/)

["Copies a string to a buffer. **Warning**: Do not use."](https://msdn.microsoft.com/en-us/library/windows/desktop/ms647490.aspx)

Please replace with `_tcscpy_s( dir, workSpaceFilePath.c_str());`.